### DISCUSSION TITLE
Read data from serial port then publish to DDS

### DISCUSSION THREAD

**Discussion by Al3x4n** [eProsima/DDS-Router#454] [Q&A] [answered]
Hello,
I'm new to the DDS technology. I have a sensor that publish data only in serial-port (100Hz/1000Hz). I know I can write application to read/write data from serial-port then publish to the DDS network but I am wondering if I can use DDS-Router as an application that bridge data from different protocols? Does It simplify the code I need to write and gives me a lot of added benefits and capabilities?
Thanks for any suggestion!

**Comment by rsanchez15** [comment, marked as answer]
Hi @Al3x4n 

Currently, the DDS Router only acts as a bridge between two DDS networks that are logically or physically isolated and independent of each other. That is, there is no protocol translation since the data received at one end are forwarded at the other end without any processing, i.e., they are not deserialized. 

The software tool that may be of interest to you, in this case, is [eProsima Integration Service,](https://github.com/eProsima/Integration-Service) which does act as a translator of other protocols to DDS. However, the case you are talking about is not supported and the product is not under active maintenance, that is, it is updated on demand.
